src/components/Onboarding/Screens/GardenMetadata.js

```diff
@@ -1,22 +1,593 @@
+          ? {
+              ...formData.links,
+              community: [
+                ...formData.links.community,
+                [{ link: '', label: '' }],
+              ],
+            }
+          : {
+              ...formData.links,
+              documentation: [
+                ...formData.links.documentation,
+                [{ link: '', label: '' }],
+              ],
+            }
+      setFormData(formData => {
+        return {
+          ...formData,
+          links: linksObject,
+        }
+      })
+    },
+    [formData.links]
+  )
+
+  const removeLink = useCallback(
+    (type, index) => {
+      const linksObject =
+        type === COMMUNITY_LINK_TYPE
+          ? {
+              ...formData.links,
+              community:
+                formData.links.community.length < 2
+                  ? [{ link: '', label: '' }]
+                  : formData.links.community.filter((_, i) => i !== index),
+            }
+          : {
+              ...formData.links,
+              documentation:
+                formData.links.documentation.length < 2
+                  ? [{ link: '', label: '' }]
+                  : formData.links.documentation.filter((_, i) => i !== index),
+            }
+
+      setFormData(formData => {
+        return {
+          ...formData,
+          links: linksObject,
+        }
+      })
+    },
+    [formData.links]
+  )
+
+  const updateLink = useCallback(
+    (type, index, updatedUrl, updatedLabel) => {
+      const linksObject =
+        type === COMMUNITY_LINK_TYPE
+          ? {
+              ...formData.links,
+              community: formData.links.community.map((link, i) =>
+                i === index ? { link: updatedUrl, label: updatedLabel } : link
+              ),
+            }
+          : {
+              ...formData.links,
+              documentation: formData.links.documentation.map((link, i) =>
+                i === index ? { link: updatedUrl, label: updatedLabel } : link
+              ),
+            }
+
+      setFormData(formData => {
+        return {
+          ...formData,
+          links: linksObject,
+        }
+      })
+    },
+    [formData.links]
+  )
+
+  const handleOnAssetAdded = useCallback((type, base64, fileExtension) => {
+    setFormData(formData => {
+      return {
+        ...formData,
+        [type]: base64,
+        [`${type}Extension`]: fileExtension,
+      }
+    })
+  }, [])
+
+  const handleOnGardenLogoTypeLoaded = useCallback(
+    (base64, fileExtension) => {
+      handleOnAssetAdded(GARDEN_LOGO_TYPE, base64, fileExtension)
+    },
+    [handleOnAssetAdded]
+  )
+
+  const handleOnGardenLogoLoaded = useCallback(
+    (base64, fileExtension) => {
+      handleOnAssetAdded(GARDEN_LOGO, base64, fileExtension)
+    },
+    [handleOnAssetAdded]
+  )
+
+  const handleOnTokenLogoLoaded = useCallback(
+    (base64, fileExtension) => {
+      handleOnAssetAdded(TOKEN_LOGO, base64, fileExtension)
+    },
+    [handleOnAssetAdded]
+  )
```

Comment:
Wonder if you consider using `handleOnAssetAdded` directly instead of creating several partially applied functions.

Reply:
this is to avoid recreating the function by using `onClick={() => handleOnAssetAdded(xxx,xxx,xx)}`